src/util.cpp

```diff
@@ -214,6 +238,24 @@ bool LogAcceptCategory(const char* category)
     return true;
 }
 
+std::string LogTimestampStr(const std::string &str)
```

Comment:
I'd suggest passing in `bool& fStartedNewLine` explicitly, so that it can be defined at the caller site. An utility function with internal state is impossible to use from different contexts.